app/legacy_lib/to_deprecated_h.rb

```diff
@@ -0,0 +1,16 @@
+# License: AGPL-3.0-or-later WITH Web-Template-Output-Additional-Permission-3.0-or-later
```

Comment:
So here's the issue: previously some of these pieces of code assumed that ActionController::Parameters would inherit from Hash. That's not the case in Rails 5.1. This works around this by giving a common interface for converting to a hash regardless of whether the variable is a hash or an ActionController::Parameters. It's not a great solution but it works.

Longer term: We should use strong params, and use ActiveModel with associated validations to handle this kind of work where we're not directly passing into a single ActiveRecord object. But you can't easily use ActiveModel with validations like an ActiveRecord object in rails 5.1 so we're not doing that. _le sigh_